pandas/tests/test_extension_pyarrow.py

```diff
@@ -0,0 +1,93 @@
+import pyarrow as pa
+
+import numpy as np
+import pandas as pd
+import pandas.util.testing as tm
+from pandas.core.extensions import ExtensionArray, ExtensionDtype
+
+
+class MyDtypeType(type):
+    pass
+
+
+class ArrowDtype(ExtensionDtype):
+    _can_hold_na = True
+    type = MyDtypeType
+    base = None
+    name = 'pa64'
+    arrow_type = pa.int64()
+
+
+class ArrowArray(ExtensionArray):
+    dtype = ArrowDtype()
+    ndim = 1
+    can_hold_na = True
+
+    def __init__(self, values):
+        data = pa.array(values)
+        assert data.type == self.dtype.arrow_type
+        self.data = data
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __len__(self):
+        return len(self.data)
+
+    @property
+    def nbytes(self):
+        return 64 * len(self)
+
+    @property
+    def shape(self):
+        return (len(self),)
+
+    def take(self, indexer, allow_fill=True, fill_value=None):
+        return type(self)(self.data.to_pandas().take(indexer))
+
+    take_nd = take
+
+    def copy(self):
+        # TODO: Jira for pa.array(pyarrow.array)
+        return pa.array(self.data.to_pandas())
+
+    def isna(self):
+        # https://github.com/apache/arrow/pull/1378
+        return pd.isna(self.data.to_pandas())
+
+    @classmethod
+    def concat_same_type(cls, to_concat):
+        return cls(np.concatenate([arr.data.to_pandas() for arr in to_concat]))
```

Comment:
You probably don't want to actually use `np.concatenate()` here, since converting to NumPy arrays can be slow and/or lossy.